File: scripts/inspect-excel.ts
import XLSX from "xlsx";
import fs from "fs";
import path from "path";

const sourceDir = "/workspaces/sae-crm-source";

function walk(dir: string): string[] {
  const results: string[] = [];

  for (const item of fs.readdirSync(dir)) {
    const fullPath = path.join(dir, item);
    const fullItemPath = path.join(dir, item);
    const stat = fs.statSync(fullItemPath);

    if (stat.isDirectory()) {
      results.push(...walk(fullItemPath));
    } else if (/\.(xlsx|xls)$/i.test(item)) {
      results.push(fullItemPath);
    }
  }

  return results;
}

const files = walk(sourceDir);

console.log("\n========================================");
console.log("SAE CRM — EXCEL SOURCE INVENTORY");
console.log("========================================");

for (const file of files) {
  console.log(`\n📘 FILE: ${path.basename(file)}`);
  console.log(`   PATH: ${file}`);

  const workbook = XLSX.readFile(file);

  console.log(`   SHEETS: ${workbook.SheetNames.length}`);

  for (const sheetName of workbook.SheetNames) {
    const sheet = workbook.Sheets[sheetName];
    const range = sheet["!ref"] || "EMPTY";

    const data = XLSX.utils.sheet_to_json<any[]>(sheet, {
      header: 1,
      defval: null,
      raw: true,
    });

    const columnCount = data.reduce(
      (max, row) => Math.max(max, row.length),
      0
    );

    const nonEmptyRows = data.filter((row) =>
      row.some((cell) => cell !== null && cell !== "")
    );

    console.log(`\n   ── SHEET: ${sheetName}`);
    console.log(`      Range: ${range}`);
    console.log(`      Rows: ${data.length}`);
    console.log(`      Columns: ${columnCount}`);
    console.log(`      Non-empty rows: ${nonEmptyRows.length}`);

    console.log("      First 5 non-empty rows:");

    nonEmptyRows.slice(0, 5).forEach((row, index) => {
      console.log(`        ${index + 1}: ${JSON.stringify(row)}`);
    });
  }
}

console.log("\n========================================");
console.log("END OF INVENTORY");
console.log("========================================\n");

File: scripts/inspect-excel.py
import os
import openpyxl

source = "/workspaces/sae-crm-source"

print("\n========================================")
print("SAE CRM — EXCEL SOURCE INVENTORY")
print("========================================")

for root, dirs, files in os.walk(source):
    for filename in sorted(files):
        if filename.lower().endswith((".xlsx", ".xlsm")):
            filepath = os.path.join(root, filename)
            print(f"\nFILE: {filename}")
            print(f"PATH: {filepath}")

            try:
                wb = openpyxl.load_workbook(filepath, read_only=True, data_only=False)

                print(f"SHEETS: {len(wb.sheetnames)}")

                for ws in wb.worksheets:
                    print(f"\n  SHEET: {ws.title}")
                    print(f"  DIMENSION: {ws.max_row} rows x {ws.max_column} columns")

                    shown = 0

                    for row in ws.iter_rows(values_only=True):
                        if any(v is not None and v != "" for v in row):
                            print("  ROW:", repr(row))
                            shown += 1
                            if shown >= 5:
                                break

                wb.close()

            except Exception as e:
                print(f"ERROR: {e}")

print("\n========================================")
print("END OF INVENTORY")
print("========================================")
